diff --git a/IDE/IAR-EWARM/Projects/CyaSSL-Lib/CyaSSL-Lib.ewp b/IDE/IAR-EWARM/Projects/CyaSSL-Lib/CyaSSL-Lib.ewp
index bc278a4..f0a1e68 100644
--- a/IDE/IAR-EWARM/Projects/CyaSSL-Lib/CyaSSL-Lib.ewp
+++ b/IDE/IAR-EWARM/Projects/CyaSSL-Lib/CyaSSL-Lib.ewp
@@ -29,8 +29,8 @@
         </option>
         <option>
           <name>Variant</name>
-          <version>20</version>
-          <state>38</state>
+          <version>21</version>
+          <state>40</state>
         </option>
         <option>
           <name>GEndianMode</name>
@@ -60,12 +60,12 @@
         </option>
         <option>
           <name>FPU</name>
-          <version>2</version>
-          <state>0</state>
+          <version>3</version>
+          <state>5</state>
         </option>
         <option>
           <name>OGCoreOrChip</name>
-          <state>0</state>
+          <state>1</state>
         </option>
         <option>
           <name>GRuntimeLibSelect</name>
@@ -87,7 +87,7 @@
         </option>
         <option>
           <name>OGLastSavedByProductVersion</name>
-          <state>6.60.1.5099</state>
+          <state>7.30.4.8186</state>
         </option>
         <option>
           <name>GeneralEnableMisra</name>
@@ -99,7 +99,7 @@
         </option>
         <option>
           <name>OGChipSelectEditMenu</name>
-          <state>MB9BF618T	Fujitsu MB9BF618T</state>
+          <state>MK70FN1M0xxx12	Freescale MK70FN1M0xxx12</state>
         </option>
         <option>
           <name>GenLowLevelInterface</name>
@@ -137,13 +137,13 @@
         </option>
         <option>
           <name>GFPUCoreSlave</name>
-          <version>20</version>
-          <state>38</state>
+          <version>21</version>
+          <state>40</state>
         </option>
         <option>
           <name>GBECoreSlave</name>
-          <version>20</version>
-          <state>38</state>
+          <version>21</version>
+          <state>40</state>
         </option>
         <option>
           <name>OGUseCmsis</name>
@@ -163,12 +163,12 @@
       <name>ICCARM</name>
       <archiveVersion>2</archiveVersion>
       <data>
-        <version>29</version>
+        <version>31</version>
         <wantNonLocal>1</wantNonLocal>
         <debug>1</debug>
         <option>
           <name>CCDefines</name>
-          <state></state>
+          <state>DEBUG_CYASSL</state>
         </option>
         <option>
           <name>CCPreprocFile</name>
@@ -229,7 +229,7 @@
         <option>
           <name>CCAllowList</name>
           <version>1</version>
-          <state>0000000</state>
+          <state>00000000</state>
         </option>
         <option>
           <name>CCDebugInfo</name>
@@ -297,6 +297,12 @@
         </option>
         <option>
           <name>CCIncludePath2</name>
+          <state>$PROJ_DIR$\..\..\..\..\..\..\MQX_4_1\lib\twrk70f120m.iar\debug</state>
+          <state>$PROJ_DIR$\..\..\..\..\..\..\MQX_4_1\lib\twrk70f120m.iar\debug\bsp\Generated_Code</state>
+          <state>$PROJ_DIR$\..\..\..\..\..\..\MQX_4_1\lib\twrk70f120m.iar\debug\bsp</state>
+          <state>$PROJ_DIR$\..\..\..\..\..\..\MQX_4_1\lib\twrk70f120m.iar\debug\psp</state>
+          <state>$PROJ_DIR$\..\..\..\..\..\..\MQX_4_1\lib\twrk70f120m.iar\debug\rtcs</state>
+          <state>$PROJ_DIR$\..\..\..\..\..\..\MQX_4_1\lib\twrk70f120m.iar\debug\mfs</state>
           <state>$PROJ_DIR$\..\..\..\..</state>
         </option>
         <option>
@@ -398,6 +404,15 @@
           <name>CCNoLiteralPool</name>
           <state>0</state>
         </option>
+        <option>
+          <name>CCOptStrategySlave</name>
+          <version>0</version>
+          <state>1</state>
+        </option>
+        <option>
+          <name>CCGuardCalls</name>
+          <state>1</state>
+        </option>
       </data>
     </settings>
     <settings>
@@ -602,6 +617,7 @@
       <data>
         <extensions></extensions>
         <cmdline></cmdline>
+        <hasPrio>0</hasPrio>
       </data>
     </settings>
     <settings>
@@ -924,7 +940,7 @@
         </option>
         <option>
           <name>IarchiveOutput</name>
-          <state>C:\ROOT\CyaSSL-Release\IAR\IDE\IAR-EWARM\Projects\CyaSSL-Lib\Debug\Exe\CyaSSL-Lib.a</state>
+          <state>C:\Work\MQX-test\externals\SSL\CyaSSL\IDE\IAR-EWARM\Projects\CyaSSL-Lib\Debug\Exe\CyaSSL-Lib.a</state>
         </option>
       </data>
     </settings>
@@ -961,7 +977,7 @@
         </option>
         <option>
           <name>Variant</name>
-          <version>20</version>
+          <version>21</version>
           <state>0</state>
         </option>
         <option>
@@ -992,7 +1008,7 @@
         </option>
         <option>
           <name>FPU</name>
-          <version>2</version>
+          <version>3</version>
           <state>0</state>
         </option>
         <option>
@@ -1069,12 +1085,12 @@
         </option>
         <option>
           <name>GFPUCoreSlave</name>
-          <version>20</version>
+          <version>21</version>
           <state>1</state>
         </option>
         <option>
           <name>GBECoreSlave</name>
-          <version>20</version>
+          <version>21</version>
           <state>1</state>
         </option>
         <option>
@@ -1095,7 +1111,7 @@
       <name>ICCARM</name>
       <archiveVersion>2</archiveVersion>
       <data>
-        <version>29</version>
+        <version>31</version>
         <wantNonLocal>1</wantNonLocal>
         <debug>0</debug>
         <option>
@@ -1161,7 +1177,7 @@
         <option>
           <name>CCAllowList</name>
           <version>1</version>
-          <state>1111111</state>
+          <state>11111110</state>
         </option>
         <option>
           <name>CCDebugInfo</name>
@@ -1330,6 +1346,15 @@
           <name>CCNoLiteralPool</name>
           <state>0</state>
         </option>
+        <option>
+          <name>CCOptStrategySlave</name>
+          <version>0</version>
+          <state>0</state>
+        </option>
+        <option>
+          <name>CCGuardCalls</name>
+          <state>1</state>
+        </option>
       </data>
     </settings>
     <settings>
@@ -1534,6 +1559,7 @@
       <data>
         <extensions></extensions>
         <cmdline></cmdline>
+        <hasPrio>0</hasPrio>
       </data>
     </settings>
     <settings>
@@ -1965,9 +1991,6 @@
       <name>$PROJ_DIR$\..\..\..\..\ctaocrypt\src\misc.c</name>
     </file>
     <file>
-      <name>$PROJ_DIR$\..\..\..\..\ctaocrypt\src\wc_port.c</name>
-    </file>
-    <file>
       <name>$PROJ_DIR$\..\..\..\..\ctaocrypt\src\pwdbased.c</name>
     </file>
     <file>
@@ -1994,6 +2017,9 @@
     <file>
       <name>$PROJ_DIR$\..\..\..\..\ctaocrypt\src\tfm.c</name>
     </file>
+    <file>
+      <name>$PROJ_DIR$\..\..\..\..\ctaocrypt\src\wc_port.c</name>
+    </file>
   </group>
 </project>
 
diff --git a/cyassl/ctaocrypt/settings.h b/cyassl/ctaocrypt/settings.h
index 29b820b..d27d796 100644
--- a/cyassl/ctaocrypt/settings.h
+++ b/cyassl/ctaocrypt/settings.h
@@ -53,7 +53,7 @@
 
 /* Uncomment next line if using PIC32MZ Crypto Engine */
 /* #define CYASSL_MICROCHIP_PIC32MZ */
-        
+
 /* Uncomment next line if using FreeRTOS */
 /* #define FREERTOS */
 
@@ -73,7 +73,8 @@
 /* #define CYASSL_LSR */
 
 /* Uncomment next line if building CyaSSL for Freescale MQX/RTCS/MFS */
-/* #define FREESCALE_MQX */
+#define FREESCALE_MQX
+#define NO_FILESYSTEM
 
 /* Uncomment next line if using STM32F2 */
 /* #define CYASSL_STM32F2 */
@@ -86,9 +87,9 @@
 
 /* Uncomment next line if building for EROAD */
 /* #define CYASSL_EROAD */
-      
+
 /* Uncomment next line if building for IAR EWARM */
-/* #define CYASSL_IAR_ARM */
+#define CYASSL_IAR_ARM
 
 /* Uncomment next line if using TI-RTOS settings */
 /* #define CYASSL_TIRTOS */
@@ -124,11 +125,11 @@
     #define NO_HC128
     #define NO_RSA
     #define NO_SESSION_CACHE
-    #define HAVE_ECC 
+    #define HAVE_ECC
 #endif
 
 
-#ifdef THREADX 
+#ifdef THREADX
     #define SIZEOF_LONG_LONG 8
 #endif
 
@@ -142,8 +143,8 @@
     #define SINGLE_THREADED
     #define CYASSL_USER_IO
     #define NO_FILESYSTEM
-#endif 
-    
+#endif
+
 #if defined(CYASSL_IAR_ARM)
     #define NO_MAIN_DRIVER
     #define SINGLE_THREADED
@@ -154,7 +155,7 @@
     #define CYASSL_USER_IO
     #define  BENCH_EMBEDDED
 #endif
-      
+
 #ifdef MICROCHIP_PIC32
     /* #define CYASSL_MICROCHIP_PIC32MZ */
     #define SIZEOF_LONG_LONG 8
@@ -271,7 +272,7 @@
 #if defined(CYASSL_LEANPSK) && !defined(XMALLOC_USER)
     #include <stdlib.h>
     #define XMALLOC(s, h, type)  malloc((s))
-    #define XFREE(p, h, type)    free((p)) 
+    #define XFREE(p, h, type)    free((p))
     #define XREALLOC(p, n, h, t) realloc((p), (n))
 #endif
 
@@ -403,7 +404,7 @@
 
     #include "SafeRTOS/heap.h"
     #define XMALLOC(s, h, type)  pvPortMalloc((s))
-    #define XFREE(p, h, type)    vPortFree((p)) 
+    #define XFREE(p, h, type)    vPortFree((p))
     #define XREALLOC(p, n, h, t) pvPortRealloc((p), (n))
 #endif
 
@@ -426,6 +427,27 @@
     #define TFM_TIMING_RESISTANT
     #define FREESCALE_K70_RNGA
     /* #define FREESCALE_K53_RNGB */
+    
+    #ifdef CYASSL_USER_IO
+           #undef CYASSL_USER_IO
+    #endif                    
+    #ifdef NO_SHA256  
+           #undef NO_SHA256           
+    #endif                 
+    #ifndef CYASSL_SHA256  
+           #define CYASSL_SHA256           
+    #endif                            
+    #ifdef NO_SHA512           
+           #undef NO_SHA512
+    #endif                 
+    #ifndef CYASSL_SHA512  
+           #define CYASSL_SHA512           
+    #endif                 
+
+    #ifndef HAVE_WEBSERVER  
+           #define HAVE_WEBSERVER           
+    #endif                         
+           
     #include "mqx.h"
     #ifndef NO_FILESYSTEM
         #include "mfs.h"
@@ -479,7 +501,7 @@
                      (CPU_CHAR *)(pstr_src), (CPU_SIZE_T)(len_max)))
     #define XSTRNCMP(pstr_1, pstr_2, len_max) \
                     ((CPU_INT16S)Str_Cmp_N((CPU_CHAR *)(pstr_1), \
-                     (CPU_CHAR *)(pstr_2), (CPU_SIZE_T)(len_max)))  
+                     (CPU_CHAR *)(pstr_2), (CPU_SIZE_T)(len_max)))
     #define XSTRSTR(pstr, pstr_srch) \
                     ((CPU_CHAR *)Str_Str((CPU_CHAR *)(pstr), \
                      (CPU_CHAR *)(pstr_srch)))
@@ -494,7 +516,7 @@
     #define XMEMMOVE XMEMCPY
 
 #if (NET_SECURE_MGR_CFG_EN == DEF_ENABLED)
-    #define MICRIUM_MALLOC    
+    #define MICRIUM_MALLOC
     #define XMALLOC(s, h, type) ((void *)NetSecure_BlkGet((CPU_INT08U)(type), \
                                  (CPU_SIZE_T)(s), (void *)0))
     #define XFREE(p, h, type)   (NetSecure_BlkFree((CPU_INT08U)(type), \
@@ -594,13 +616,13 @@
     #endif
 
     #if (SSL_CFG_USER_RNG_SEED_EN == DEF_ENABLED)
-        #define NO_DEV_RANDOM   
+        #define NO_DEV_RANDOM
     #else
         #undef  NO_DEV_RANDOM
     #endif
 
     #if (SSL_CFG_USER_IO_EN == DEF_ENABLED)
-        #define CYASSL_USER_IO   
+        #define CYASSL_USER_IO
     #else
         #undef  CYASSL_USER_IO
     #endif
@@ -714,8 +736,8 @@
         #define CYASSL_GENERAL_ALIGNMENT  4
     #elif defined(FREESCALE_MMCAU)
         #define CYASSL_GENERAL_ALIGNMENT  CYASSL_MMCAU_ALIGNMENT
-    #else 
-        #define CYASSL_GENERAL_ALIGNMENT  0 
+    #else
+        #define CYASSL_GENERAL_ALIGNMENT  0
     #endif
 #endif
 
@@ -732,7 +754,7 @@
 
 
 /* Place any other flags or defines here */
-
+      
 
 #ifdef __cplusplus
     }   /* extern "C" */
